src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/NteractKernelExtensionTests.cs

```diff
@@ -0,0 +1,190 @@
+
+            formatted.Should().Match("*getJsLoader*");
+        }
+
+        [Fact]
+        public async Task it_can_loads_script_from_uri_and_specify_context()
+        {
+            using var kernel = new CompositeKernel();
+
+            var kernelExtension = new NteractKernelExtension();
+            DataExplorerExtensions.Settings.UseUri("https://a.cdn.url/script.js", "2.2.2");
+            await kernelExtension.OnLoadAsync(kernel);
+
+            var data = new[]
+            {
+                new {Type="orange", Price=1.2},
+                new {Type="apple" , Price=1.3},
+                new {Type="grape" , Price=1.4}
+            };
+
+
+            var formatted = data.ToTabularJsonString().ToDisplayString(HtmlFormatter.MimeType);
+
+            formatted.Should().Match("*'context': '2.2.2'*");
+        }
+
+        [Fact]
+        public async Task uri_is_quoted()
+        {
+            using var kernel = new CompositeKernel();
+
+            var kernelExtension = new NteractKernelExtension();
+            DataExplorerExtensions.Settings.UseUri("https://a.cdn.url/script.js");
+            await kernelExtension.OnLoadAsync(kernel);
+
+            var data = new[]
+            {
+                new {Type="orange", Price=1.2},
+                new {Type="apple" , Price=1.3},
+                new {Type="grape" , Price=1.4}
+            };
+
+
+            var formatted = data.ToTabularJsonString().ToDisplayString(HtmlFormatter.MimeType);
+
+            formatted.Should().Match("*'https://a.cdn.url/script'*");
+        }
+
+        [Fact]
+        public async Task uri_extension_is_removed()
+        {
+            using var kernel = new CompositeKernel();
+
+            var kernelExtension = new NteractKernelExtension();
+            DataExplorerExtensions.Settings.UseUri("https://a.cdn.url/script.js");
+            await kernelExtension.OnLoadAsync(kernel);
+
+            var data = new[]
+            {
+                new {Type="orange", Price=1.2},
+                new {Type="apple" , Price=1.3},
+                new {Type="grape" , Price=1.4}
+            };
+
+
+            var formatted = data.ToTabularJsonString().ToDisplayString(HtmlFormatter.MimeType);
+
+            formatted.Should().NotContain("'https://a.cdn.url/script.js'");
+        }
+
+        [Fact]
+        public async Task can_specify_cacheBuster()
+        {
+            using var kernel = new CompositeKernel();
+
+            var kernelExtension = new NteractKernelExtension();
+            DataExplorerExtensions.Settings.UseUri("https://a.cdn.url/script.js", cacheBuster:"XYZ");
+            await kernelExtension.OnLoadAsync(kernel);
+
+            var data = new[]
+            {
+                new {Type="orange", Price=1.2},
+                new {Type="apple" , Price=1.3},
+                new {Type="grape" , Price=1.4}
+            };
+
+
+            var formatted = data.ToTabularJsonString().ToDisplayString(HtmlFormatter.MimeType);
```

Comment:
Why not `data.ToDisplayString(TabularDataFormatter.MimeType)`?

Reply:
I am testing the tabluarJsonString formatter, the open producing the full html code